addons/controllers/cpi/vspherecpiconfig_utils.go

```diff
@@ -84,27 +71,9 @@ func (r *VSphereCPIConfigReconciler) mapCPIConfigToDataValuesNonParavirtual( //
 		d.NoProxy = cluster.Annotations[pkgtypes.NoProxyConfigAnnotation]
 	}
 
-	// derive nsxt related configs from cluster variable
-	d.Nsxt.PodRoutingEnabled = r.tryParseClusterVariableBool(cluster, NsxtPodRoutingEnabledVarName)
```

Comment:
Why are nsxt values being removed?

Reply:
These values are not required by cluster-api, so they will never be cluster variables.

Reply:
The file you referenced is for backwards compatibility with legacy configuration. Going forward we expect these to be provided as cluster variables, if they are missing then we have a gap in the cluster class definition. cc @vuil @lubronzhan

Reply:
>Going forward we expect these to be provided as cluster variables
If that's the case @DanielXiao could you revert this part of the change? Thanks